File: app/src/test/java/eu/kanade/tachiyomi/TestApp.java
package eu.kanade.tachiyomi;

import eu.kanade.tachiyomi.injection.component.DaggerAppComponent;
import eu.kanade.tachiyomi.injection.module.AppModule;
import eu.kanade.tachiyomi.injection.module.TestDataModule;

public class TestApp extends App {

    @Override
    protected DaggerAppComponent.Builder prepareAppComponent() {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .dataModule(new TestDataModule());
    }

    @Override
    protected void setupEventBus() {
        // Do nothing
    }

    @Override
    protected void setupAcra() {
        // Do nothing
    }
}

File: app/src/test/java/eu/kanade/tachiyomi/TestApp.kt
package eu.kanade.tachiyomi

import eu.kanade.tachiyomi.injection.component.AppComponent
import eu.kanade.tachiyomi.injection.component.DaggerAppComponent
import eu.kanade.tachiyomi.injection.module.AppModule
import eu.kanade.tachiyomi.injection.module.TestDataModule

open class TestApp : App() {

    override fun createAppComponent(): AppComponent {
        return DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .dataModule(TestDataModule())
                .build()
    }

    override fun setupAcra() {
        // Do nothing
    }
}
